src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt

```diff
@@ -111,6 +113,19 @@ class AddressService(
     return casCheckRequest.toSummary()
   }
 
+  @Transactional
+  fun deleteAddressCheckRequest(prisonNumber: String, requestId: Long) {
+    val curfewAddressCheckRequest =
+      curfewAddressCheckRequestRepository.findByIdOrNull(requestId)
+        ?: throw EntityNotFoundException("Cannot find standard address check request with id: $requestId")
+
+    if (curfewAddressCheckRequest.assessment.offender.prisonNumber != prisonNumber) {
+      throw HttpClientErrorException(HttpStatus.UNAUTHORIZED, "Standard address check request id: $requestId is not linked to offender with prison number: $prisonNumber")
```

Comment:
**suggestion (non-blocking):** maybe this could be a 404 with the same message? 403 will be handled by the frontend to log the user out

Reply:
ahh ok, will do it now so it doesn't get forgotten